src/index.ts

```diff
@@ -46,6 +46,31 @@ const getMessagesFromStore = ({ storageType, storageKey }: StorageParams) => {
   }
 };
 
+const validateWebSocketUrl = (rawUrl: string): URL => {
+  let url: URL;
+  try {
+    // Alternatively, we can also check with URL.canParse(), but since we need
+    // the URL object anyway to validate the protocol, we go ahead and parse it
+    // here.
+    url = new URL(rawUrl);
+  } catch (e) {
+    // TypeError, as specified by WHATWG URL Standard:
+    // https://url.spec.whatwg.org/#url-class (see constructor steps)
+    if (!(e instanceof TypeError)) {
+      throw e;
+    }
+    const { message } = e;
+    throw new Error(`The WebSocket URL is not valid: ${message}`);
+  }
+  const { protocol } = url;
+  if (protocol === "wss:" || protocol == "ws:") {
```

Comment:
Spotted a `==` instead of a `===` here.

An alternative way to execute the logic could be an array includes call:

```javascript
['ws://', 'wss://'].includes(protocol)
```